patroni/postgresql.py

```diff
@@ -724,16 +699,20 @@ def is_leader(self):
             raise PostgresConnectionException(str(e))
 
     def is_running(self):
-        if not (self._version_file_exists() and os.path.isfile(self._postmaster_pid)):
-            # XXX: This is dangerous in case somebody deletes the data directory while PostgreSQL is still running.
-            return False
-
-        pidfile = self.read_pid_file()
-        return self._is_postmaster_pid_running(int_or_none(pidfile.get('pid')),
-                                               start_time=int_or_none(pidfile.get('start_time')))
-
-    @_update_postmaster_cached_info
-    def read_pid_file(self):
+        """Returns PostmasterProcess if one is running on the data directory or None. If most recently seen process
+        is running udpates the cached process based on pid file."""
+        if self._postmaster_proc:
+            if self._postmaster_proc.is_running():
+                # TODO: is this check useful?
```

Comment:
This check came from https://github.com/postgres/postgres/blob/master/src/bin/pg_ctl/pg_ctl.c#L277

Since we are caching `PostmasterProcess` object, it is useless in this place.